infrastructure/src/main/java/org/corfudb/infrastructure/SequencerServerCache.java

```diff
@@ -34,8 +30,10 @@
      * <p>
      * a cache of recent conflict keys and their latest global-log position.
      */
-    private final Cache<ConflictTxStream, Long> conflictCache;
-
+    private final Hashtable<ConflictTxStream, Long> cacheHashtable;
```

Comment:
what was the motivation for replacing caffeine cache by hasttable?

Reply:
Why this conversation is resolved?

Reply:
I have explained before, we have seen overflown issues with caffein cache. The eviction mechanism like LRU etc is not feasible for our use case.

Reply:
Actually `size` based eviction should work. 
Instead of `size` you can use global addresses.
https://github.com/ben-manes/caffeine/wiki/Eviction

Reply:
@xnull 
The cache map maps conflict keys (stream id + key) to versions (long), illustrated below:

Conflict Key | ck1 | ck2 | ck3 | ck4
Version        | v1   | v1   |  v2  |  v3

Consider the case where we need to insert a new conflict key (ck), but the cache is full, then we need to evict the oldest conflict keys. In the example above, we can't just evict ck1, need to evict all keys that map to v1, so we need to evict ck1 and ck2, this eviction policy is FIFO on the version number. Notice that we also can't evict ck3 before the keys for v1, that's because it will create holes in the resolution window and can lead to incorrect resolutions. 

How can we use the caffeine cache to achieve this behavior ?

Note that this is a sliding window on the versions, where a version can map to multiple keys, so we also need to maintain the begining of the window which is the `maxConflictWildcard` variable.


@xiaoqin2012 Please capture this in the class comments, because its not obvious to everyone why we need to have this new data structure.